interpreter/cling/lib/Interpreter/IncrementalParser.cpp

```diff
@@ -158,7 +181,10 @@ namespace {
           return; // ignore!
         }
       }
-      m_PrevClient.HandleDiagnostic(DiagLevel, Info);
+      if (auto C = m_Target.getPointer())
+        C->HandleDiagnostic(DiagLevel, Info);
+      else
+        m_PrevClient.HandleDiagnostic(DiagLevel, Info);
```

Comment:
Should the diag be emitted to both? (I'm asking for a rationale, and if you know the answer maybe add a comment to clarify?)

Reply:
As discussed in the documentation for `FilteringDiagConsumer`, this class acts as a proxy for some other `DiagnosticConsumer`, filtering many warnings derived of incremental compilation.

In principle, for simplicity, we preserve the old behavior of delivering diagnostics to just one consumer (that is why we don't emit to both), but we allow the "sink" to be changed.
Note, however, that consumers might update their internal state in calls to, e.g. `BeginSourceFile()` or `EndSourceFile()` (actually, `TextDiagnosticPrinter` is an example of this), so in order to be able to restore the original consumer, we need to keep forwarding these calls.

Reply:
Is that worth a comment? (Like copy&paste what you wrote as response to my question?)

Reply:
I can add it.  :-)